Allow whitespace markers for verbatim text: ´|<`, `|>`, and `|<>`

UweKubosch opened the pull request:
Fixes #882

UweKubosch commented:
Looks good?

minad commented on test/literate/TESTS.md:
```diff
@@ -27,6 +27,22 @@ renders as
+
+~~~ slim
+|< Text with leading whitespace.
+|  Text with leading whitespace.
```
Maybe also add tests here with more than a single space?

minad commented:
Looks good at first sight. I will test this soon.

UweKubosch commented:
Hmm, yes, that is a use case we need to decide on.  For compatibility `|<tag>` needs to work, right?  I can add it as a special case.

UweKubosch commented:
I added a test case with a verbatim text starting with a tag without a dividing space, and I changed the parser to handle it as before.  This means that verbatim text with pipe and whitespace markers requires a dividing space before the content.

It think it makes sense and handles backwards compatibility.

I am not sure I have caught all variants of indentation, but all existing tests are green.  What might be missing is handling of indentation with whitespace markers.

minad commented:
Thanks. I let this rest for a few days and take another closer look.

> What might be missing is handling of indentation with whitespace markers.

Yes, can you please add tests for that too?

UweKubosch commented:
> Thanks. I let this rest for a few days and take another closer look.
> 
> > What might be missing is handling of indentation with whitespace markers.
> 
> Yes, can you please add tests for that too? Maybe we also need tests for `|><`? Do we test this for the other line indicators too? In any case it makes sense to add more tests.

OK, added test case for both `|><` and indented block with whitespace markers.  Hope this makes it complete.
